src/common/darktable.c

```diff
@@ -1073,17 +1089,22 @@ int dt_init(int argc, char *argv[], const gboolean init_gui, const gboolean load
   //  2 mipmap size
   //  3 opencl available
   static int fractions[20] = {
-      512,   32, 128, 600,  // default
+      512,   32, 128, 700,  // default
         0,    0,  16,   0,  // mini
-      128,   16,  64, 256,  // small
-      700,   64, 128, 750,  // large
-    16384, 1024, 128, 900   // unrestricted
+      128,   16,  64, 400,  // small
+      700,   64, 128, 900,  // large
+    16384, 1024, 128, 1024  // unrestricted
   };
+  // Allow the settings for each performance level to be changed via darktablerc
+  check_resourcelevel("resource_default", fractions, 0);
+  check_resourcelevel("resource_small", fractions, 2);
+  check_resourcelevel("resource_large", fractions, 3);
+  check_resourcelevel("resource_unrestricted", fractions, 4);
 
   darktable.dtresources.fractions = fractions;
-  darktable.dtresources.total_memory = _get_total_memory();
-  darktable.dtresources.mipmap_memory = _get_mipmap_size();
+  darktable.dtresources.total_memory = _get_total_memory() * 1024lu;
```

Comment:
Wouldn't it be better the keep the responsibility of returning the number of bytes with `_get_total_memory`, and move the multiplication there?

Reply:
Yes, that would have possible too of course.